core/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -42,134 +46,212 @@
-            boolean satisfied = in.readBoolean();
-            conditions.add(new AbstractMap.SimpleEntry<>(condition, satisfied));
-        }
-        conditionStatus = conditions;
-        dryRun = in.readBoolean();
-        rolledOver = in.readBoolean();
-        acknowledged = in.readBoolean();
-        shardsAcked = in.readBoolean();
+        int responsesSize = in.readVInt();
+        for (int j = 0; j < responsesSize; j++) {
+            final SingleAliasRolloverResponse response = new SingleAliasRolloverResponse();
+            response.oldIndex = in.readString();
+            response.newIndex = in.readString();
+            int conditionSize = in.readVInt();
+            Set<Map.Entry<String, Boolean>> conditions = new HashSet<>(conditionSize);
+            for (int i = 0; i < conditionSize; i++) {
+                String condition = in.readString();
+                boolean satisfied = in.readBoolean();
+                conditions.add(new AbstractMap.SimpleEntry<>(condition, satisfied));
+            }
+            response.conditionStatus = conditions;
+            response.dryRun = in.readBoolean();
+            response.rolledOver = in.readBoolean();
+            response.acknowledged = in.readBoolean();
+            response.shardsAcked = in.readBoolean();
+            responses.add(response);
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeString(oldIndex);
-        out.writeString(newIndex);
-        out.writeVInt(conditionStatus.size());
-        for (Map.Entry<String, Boolean> entry : conditionStatus) {
-            out.writeString(entry.getKey());
-            out.writeBoolean(entry.getValue());
-        }
-        out.writeBoolean(dryRun);
-        out.writeBoolean(rolledOver);
-        out.writeBoolean(acknowledged);
-        out.writeBoolean(shardsAcked);
+        out.writeVInt(responses.size());
+        for (SingleAliasRolloverResponse response: responses) {
+            out.writeString(response.oldIndex);
+            out.writeString(response.newIndex);
+            out.writeVInt(response.conditionStatus.size());
+            for (Map.Entry<String, Boolean> entry : response.conditionStatus) {
+                out.writeString(entry.getKey());
+                out.writeBoolean(entry.getValue());
+            }
+            out.writeBoolean(response.dryRun);
+            out.writeBoolean(response.rolledOver);
+            out.writeBoolean(response.acknowledged);
+            out.writeBoolean(response.shardsAcked);
+        }
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+    public XContentBuilder toXContent(final XContentBuilder builder, Params params) throws IOException {
+        if (responses.isEmpty() && failures.isEmpty()) {
+            throw new IllegalStateException("Response for at least 1 alias is expected");
+        }
```

Comment:
I think this is already checked in the ctor, no need to do it again here I think (if it wasn't for this stupid dummy ctor with empty arguments). I think the current RolloverResponse would also break here if the empty constructor was used.

Reply:
Empty arguments constructor is used together with ```readFrom``` (which doesn't prohibit to have 0 ```SingleAliasRolloverResponse```). So theoretically it is possible to have empty responses.
But if we will make backward compatible ```readFrom``` and ```writeTo``` then this check in ```toXContentBuilder``` will become redudant.